### DISCUSSION TITLE
Transform multiple create table statements into excel documentation

### DISCUSSION THREAD

**Discussion by nicoDs96** [xnuinside/simple-ddl-parser#164] [Show and tell]
Hi, 
one of my task of the week was to document into an excel file all the tables of the database. Using my DB client ([DBeaver](https://dbeaver.io/)) I was able to create a giant .sql file with all the create tables. 
Then I should put one by one all the table name, the schema, the fields and the fileds' type, the source etc. into an excel file. 
Here is where I found this amazing library that made me complete the task in 20 minutes instead of days. 
Let's see how...
I have DDL.sql
```sql
CREATE TABLE schema.table1 (
	field1 type NOT NULL,
	field2 type NULL,
	....);


-- some auto-generated comment here
CREATE TABLE schema.table2 (
	field1 type NOT NULL,
	field2 type NULL,
	....);

-- and so on for thousand of tables...
```
I shoud document those tables into an excel file like:  
|table_name|schema|field|type|...|
|----|----|----|----|----|
|table1 |schema|field1 |type1|...|
|table1 |schema|field2 |type1|...|
|...|||||
|table2 |schema|field1 |type1|...|
|table2 |schema|field2 |type1|...|
|...|||||

I ended up doing this task with a tiny python script (named ddl2tsv.py):
```python
from simple_ddl_parser import DDLParser
with open('DDL.sql', 'r') as file:
    ddl_statements= file.read()
    parse_results = DDLParser(ddl_statements).run()
    for pr in parse_results:
	    for col in pr['columns']:
		    type = col['type'] if col["size"] is None else f'{col["type"]}({col["size"]})'
		    print(f'{pr["table_name"]}\t{pr["schema"]}\t{col["name"]}\t{type}')
```
and then
```bash 
python ddl2tsv.py > doc.tsv
```
Now I only need to open the tsv file and copy paste the columns into the doc file and do some small adjustments.

In conclusion, thank you guys for this amazing tool you provided to us!

**Comment by xnuinside** [comment]
@nicoDs96 this is an amazing feedback that inspire me a lot to continue work on it and deliver new versions. But I’m alone )))) peoples who opens the issues helps a lot because they provides data samples and report about problems. But I have a lack of time to deliver fixes faster :( but thank you for the amazing feedback!
